### PR TITLE
Updated Data Gen 2 authorization rules to use the new syntax

### PR DESCRIPTION
**Pull request by renebrandel**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/gen2/build-a-backend/data/customize-authz/signed-in-user-data-access/index.mdx [hunk 1]
```diff
@@ -64,7 +64,7 @@ const schema = a.schema({
     .model({
       content: a.string(),
     })
-    .authorization([a.allow.private('identityPool')]),
+    .authorization(allow => [allow.authenticated('identityPool')]),
```
Comment (HuiSF) [inline src/pages/gen2/build-a-backend/data/customize-authz/signed-in-user-data-access/index.mdx:67]:
```suggestion
    .authorization(allow => [allow.authenticated('iam')]),
```

**Review by HuiSF — APPROVED** [review]

File: src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx [hunk 2]
```diff
@@ -55,7 +55,7 @@ const schema = a
   })
   // highlight-start
   .authorization([
-    a.allow.resource(functionWithDataAccess).to(['query', 'listen'])
+    allow.resource(functionWithDataAccess).to(['query', 'listen'])
```
Comment (johnpc) [inline src/pages/gen2/build-a-backend/data/customize-authz/grant-lambda-function-access-to-api/index.mdx:58]: should this be `.authorization(allow => [allow.resource(functionWithDataAccess).to(['query', 'listen'])])`?

**Review by reesscot — APPROVED** [review]

**Review by johnpc — APPROVED** [review]

File: src/pages/gen2/build-a-backend/data/customize-authz/public-data-access/index.mdx [hunk 3]
```diff
@@ -57,7 +57,7 @@ const schema = a.schema({
     .model({
       content: a.string(),
     })
-    .authorization([a.allow.public('identityPool')]),
+    .authorization(allow => [allow.guest()]),
```
Comment (dzhan-aws) [inline src/pages/gen2/build-a-backend/data/customize-authz/public-data-access/index.mdx:60]: For line 52 above, do we need to update the part about `identityPool`?

File: src/pages/gen2/build-a-backend/data/customize-authz/signed-in-user-data-access/index.mdx [hunk 4]
```diff
@@ -64,7 +64,7 @@ const schema = a.schema({
     .model({
       content: a.string(),
     })
-    .authorization([a.allow.private('identityPool')]),
+    .authorization(allow => [allow.authenticated('iam')]),
```
Comment (dzhan-aws) [inline src/pages/gen2/build-a-backend/data/customize-authz/signed-in-user-data-access/index.mdx:67]: Ditto, do we need to change line 58 around `identityPool`?

File: src/pages/gen2/build-a-backend/data/customize-authz/public-data-access/index.mdx (hunk 3 above)
Comment (renebrandel) [reply src/pages/gen2/build-a-backend/data/customize-authz/public-data-access/index.mdx:60]: Nope not needed. guest == public('identityPoo'). publicApiKey === public()

File: src/pages/gen2/build-a-backend/data/customize-authz/signed-in-user-data-access/index.mdx (hunk 4 above)
Comment (renebrandel) [reply src/pages/gen2/build-a-backend/data/customize-authz/signed-in-user-data-access/index.mdx:67]: this is a QA issue. Already filed a separate bug.

**Review by dzhan-aws — APPROVED** [review]

**Review by hdworld11 — APPROVED** [review]

**Pull request merged**